railties/test/application/configuration_test.rb

```diff
@@ -1727,6 +1727,7 @@ def index
       app "development"
 
       assert_equal "bar", Rails.application.config.my_custom_config[:foo]
+      assert_equal "bar", Rails.application.config.my_custom_config["foo"]
```

Comment:
I would rather test deprecated behaviour in deprecation tests only.

Reply:
I have noted in the `Other Information` section of my PR that this is *not* deprecated behaviour.

> Note that the top-level access to the `OrderedOptions` object returned from `config_for` has indifferent access semantics due to the nature of how `OrderedOptions` works.

Since, `my_custom_config` is an instance of `OrderedOptions`, top-level key access is indifferent. Thus, this test is asserting that that behaviour is not changed.

I could put an `assert_not_deprecated` around the string access assertion if it is going to make it more readable and understandable, though.

Reply:
Ah, my mistake. Leaving it as-is if we don't expect this to be deprecated sounds good to me 👍